pallets/delegation/src/lib.rs

```diff
@@ -421,24 +434,56 @@ pub mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
-	// Calculate the hash of all values of a delegation transaction
-	fn calculate_hash(
+	// Calculates the hash of all values of a delegation creation transaction.
+	fn calculate_delegation_creation_hash(
```

Comment:
For the first point, the function is used only to re-create the information that the delegatee signs upon delegation creation. I agree it does refer to the delegation content, but it is used only for creations. And it is not even the hash of the whole delegation, but only what is needed to be protected when creating a new delegation.

For the second point, we have removed the root_id from the extrinsic parameter, so the submitter is not required to know what `hierarchy_id` the node will end up into, right? Otherwise are you suggesting we should still include the `root_id` in the hash but not include it as a parameter in the extrinsic call?

Reply:
> For the first point, the function is used only to re-create the information that the delegatee signs upon delegation creation. I agree it does refer to the delegation content, but it is used only for creations. And it is not even the hash of the whole delegation, but only what is needed to be protected when creating a new delegation.

The signature is used to ensure the delegation is created with consent of the receiver. i would feel better if the rootId is included.
 
> For the second point, we have removed the root_id from the extrinsic parameter, so the submitter is not required to know what `hierarchy_id` the node will end up into, right? Otherwise are you suggesting we should still include the `root_id` in the hash but not include it as a parameter in the extrinsic call?

The submitter is required to know to which delegation hierarchy the new node will be added to. But the submitter is someone else. The signature is done by the receiver of the delegation. And the receiver needs to be conscious about which delegation hierarchy he will be added to.

Reply:
Re-added `root_id` in the computation of the delegation creation hash: https://github.com/KILTprotocol/mashnet-node/pull/226/commits/310cad8bc19ba5887a722660543ab4e0b8d6ce17.